src/provider/model/database/Database.ts

```diff
@@ -0,0 +1,69 @@
+import {injectable} from 'inversify';
+import Dexie from 'dexie';
+
+import {StoredSettings} from '../StoredSettings';
+import {StoredNotification} from '../StoredNotification';
+import {DeferredPromise} from '../../common/DeferredPromise';
+import {AsyncInit} from '../../controller/AsyncInit';
+
+import {Collection} from './Collection';
+
+export const enum CollectionMap {
+    NOTIFICATIONS = 'notifications',
+    SETTINGS = 'settings'
+}
+
+export type Collections = {
+    [CollectionMap.NOTIFICATIONS]: StoredNotification;
+    [CollectionMap.SETTINGS]: StoredSettings;
+};
+
+@injectable()
+export class Database extends AsyncInit {
+    private _database: Dexie;
+    private _collections: Map<CollectionMap, Collection<any>>;
+    private _databaseReadyPromise: DeferredPromise<void>;
+
+    constructor () {
+        super();
+
+        this._database = new Dexie('Notifications_Service');
+        this._collections = new Map();
+        this._databaseReadyPromise = new DeferredPromise();
+
+        this._database.version(1).stores({
+            notifications: '&id',
+            settings: '&id'
+        });
+
+        this.createCollections(this._database.tables);
+
+        this._database.open().then(()=>{
+            this._databaseReadyPromise.resolve();
+        });
+    }
+
+    protected async init() {
```

Comment:
Return type

Reply:
Done! Although to note I see that many other files do not have a return type on this init